<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Changelog • biodosetools</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><link href="../deps/Nunito_Sans-0.4.2/font.css" rel="stylesheet"><link href="../deps/Fira_Code-0.4.2/font.css" rel="stylesheet"><link href="../deps/Nunito-0.4.2/font.css" rel="stylesheet"><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Changelog"><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">biodosetools</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">3.6.0.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <h6 class="dropdown-header" data-toc-skip>Dicentrics</h6>
    <a class="dropdown-item" href="../articles/dicent-fitting.html">Dicentrics dose-effect fitting</a>
    <a class="dropdown-item" href="../articles/dicent-estimation.html">Dicentrics dose estimation</a>
    <div class="dropdown-divider"></div>
    <h6 class="dropdown-header" data-toc-skip>Translocations</h6>
    <a class="dropdown-item" href="../articles/trans-fitting.html">Translocations dose-effect fitting</a>
    <a class="dropdown-item" href="../articles/trans-estimation.html">Translocations dose estimation</a>
  </div>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/biodosetools-team/biodosetools/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-news">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Changelog</h1>
      <small>Source: <a href="https://github.com/biodosetools-team/biodosetools/blob/HEAD/NEWS.md" class="external-link"><code>NEWS.md</code></a></small>
    </div>

    <div class="section level2">
<h2 class="pkg-version" data-toc-text="3.6.0.9000" id="biodosetools-3609000">biodosetools 3.6.0.9000<a class="anchor" aria-label="anchor" href="#biodosetools-3609000"></a></h2>
<div class="section level3">
<h3 id="improvements-3-6-0-9000">Improvements<a class="anchor" aria-label="anchor" href="#improvements-3-6-0-9000"></a></h3>
<ul><li>Updated LQ and L formulas to use lambda instead of Y for yield in <code>parse_model_formula()</code>, <code>list_fitting_formulas()</code>, and help dialogues.</li>
<li>Removed unnecessary {base} namespace in <code><a href="https://rdrr.io/r/base/solve.html" class="external-link">solve()</a></code> call.</li>
<li>Removed legacy <code>gardner_confidence_table</code> data (deprecated in <a href="https://github.com/biodosetools-team/biodosetools/commit/585e7b4b6f42e66965e3b0f6a271dbaea7b7cf28" class="external-link">585e7b4</a>).</li>
<li>Removed unused files in <code>data-raw</code>.</li>
<li>Updated <code>get_cov_ZIP_ML()</code> to match expressions on manuscript, introducing minor calculation optimisation.</li>
<li>Updated CITATION file.</li>
<li>Updated vignettes.</li>
</ul></div>
<div class="section level3">
<h3 id="bug-fixes-3-6-0-9000">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-3-6-0-9000"></a></h3>
<ul><li>Return case data <code>as_tibble()</code> in <code><a href="../reference/calculate_aberr_table.html">calculate_aberr_table()</a></code> for consistency with count data.</li>
<li>Minor formatting fix in <code><a href="https://cli.r-lib.org/reference/cli_alert.html" class="external-link">cli::cli_alert_info()</a></code> call.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="3.6.0" id="biodosetools-360">biodosetools 3.6.0<a class="anchor" aria-label="anchor" href="#biodosetools-360"></a></h2><p class="text-muted">CRAN release: 2022-01-27</p>
<ul><li>Major rewrite of <code><a href="../reference/plot_estimated_dose_curve.html">plot_estimated_dose_curve()</a></code> function.</li>
<li>Delta method calculations are now all done via <code><a href="https://rdrr.io/pkg/msm/man/deltamethod.html" class="external-link">msm::deltamethod()</a></code>.</li>
<li>Added progress and alert notifications to UI.</li>
<li>Added {markdown} as suggested dependency.</li>
<li>Added {cli} as dependency.</li>
<li>Tweaked references section in {pkgdown} site to only include exported functions organised by topic/usage.</li>
<li>Use consistently genomic conversion factor or <code>genome_factor</code> across all functions, help dialogues, and reports.</li>
<li>Added R-CMD check GitHub Action.</li>
<li>Added vignettes.</li>
</ul><div class="section level3">
<h3 id="new-features-3-6-0">New features<a class="anchor" aria-label="anchor" href="#new-features-3-6-0"></a></h3>
<ul><li>Return <code>conf_int</code> in <code>estimate_*()</code> family of functions.</li>
<li>Parse <code>est_full_doses$type</code> in <code><a href="../reference/plot_estimated_dose_curve.html">plot_estimated_dose_curve()</a></code> function so unused assessments are not shown in legend.</li>
<li>Added <code>match_names()</code> function to match est_doses list names when calling <code><a href="../reference/plot_estimated_dose_curve.html">plot_estimated_dose_curve()</a></code>.</li>
<li>Updated <code><a href="../reference/plot_estimated_dose_curve.html">plot_estimated_dose_curve()</a></code> to automatically build <code>est_full_doses</code> object from list of dose estimation results, added <code>parse_conf_int_text()</code> auxiliary function.</li>
</ul></div>
<div class="section level3">
<h3 id="bug-fixes-3-6-0">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-3-6-0"></a></h3>
<ul><li>Correct dispersion index value on <code><a href="../reference/estimate_whole_body_delta.html">estimate_whole_body_delta()</a></code> when there’s no aberrations.</li>
<li>Fixed issue in missing count data in DOCX fitting report.</li>
<li>Removed double calculation of <code>cov_est</code> in <code>estimate_partial_dolphin()</code>.</li>
<li>Wrapped mixed Poisson model dose estimation in <code><a href="https://rdrr.io/r/base/try.html" class="external-link">try()</a></code> to ensure convergence (up to 5 tries).</li>
<li>Mixed Poisson model <code><a href="../reference/estimate_hetero_mixed_poisson.html">estimate_hetero_mixed_poisson()</a></code> uses exact delta methods for dose and fraction irradiated, as described in paper by Pujol et al. (2016) <a href="https://doi.org/10.1667/RR14145.1" class="external-link uri">https://doi.org/10.1667/RR14145.1</a>.</li>
<li>Fixed <code>gamma</code> and <code>gamma_error</code> parsing when calling <code><a href="../reference/estimate_hetero_mixed_poisson.html">estimate_hetero_mixed_poisson()</a></code> in <code>mod_estimation_results_server()</code>.</li>
<li>Wrap unicode characters (<code>\uxxxx</code>) in <code><a href="https://rlang.r-lib.org/reference/as_utf8_character.html" class="external-link">rlang::as_utf8_character()</a></code> to avoid “unable to translate to native encoding” warning on Windows.</li>
</ul></div>
<div class="section level3">
<h3 id="new-functions-3-6-0">New functions<a class="anchor" aria-label="anchor" href="#new-functions-3-6-0"></a></h3>
<ul><li>
<code><a href="../reference/get_deltamethod_std_err.html">get_deltamethod_std_err()</a></code>: auxiliary function to wrap all <code><a href="https://rdrr.io/pkg/msm/man/deltamethod.html" class="external-link">msm::deltamethod()</a></code> calls.</li>
</ul></div>
<div class="section level3">
<h3 id="improvements-3-6-0">Improvements<a class="anchor" aria-label="anchor" href="#improvements-3-6-0"></a></h3>
<ul><li>Fixed order of output case data columns in <code><a href="../reference/calculate_aberr_table.html">calculate_aberr_table()</a></code> so they match the final output in Shiny.</li>
<li>Renamed <code>count-data-IAEA.csv</code> to <code>count-data-barquinero-1995.csv</code> from Barquinero et al. (1995) <a href="https://doi.org/10.1016/0027-5107(94)00150-4" class="external-link uri">https://doi.org/10.1016/0027-5107(94)00150-4</a>.</li>
<li>Added <code>count-data-rodriguez-2004.csv</code> for total translocations from Rodriguez et al. (2004) <a href="https://doi.org/10.1667/RR3198" class="external-link uri">https://doi.org/10.1667/RR3198</a>.</li>
<li>Updated translocations fitting unit tests to use data from Rodriguez et al. (2004) <a href="https://doi.org/10.1667/RR3198" class="external-link uri">https://doi.org/10.1667/RR3198</a>.</li>
<li>Rewrote <code><a href="../reference/calculate_aberr.html">calculate_aberr_power()</a></code> to remove {purrr} dependency and reduce computation time by 4.</li>
<li>Replaced <code><a href="https://rdrr.io/r/base/message.html" class="external-link">message()</a></code> calls with appropriate <code>cli::cli_*()</code> calls.</li>
<li>Split quasi-Poisson from automatic fitting in <code><a href="../reference/fit_glm_method.html">fit_glm_method()</a></code>, as it is otherwise confusing.</li>
<li>Updated <code>\sigma</code> to <code>\hat{\sigma}</code> in <code>fix_count_data_names()</code> and respective tests.</li>
<li>Updated unit tests to test individual values both in dose-effect fitting and dose estimation.</li>
<li>Use <code>*-fitting-results-*.rds</code> instead of <code>*-fitting-data-*.rds</code> when exporting RDS objects in <code>mod_fitting_results_server()</code>.</li>
</ul></div>
<div class="section level3">
<h3 id="ui-improvements-3-6-0">UI Improvements<a class="anchor" aria-label="anchor" href="#ui-improvements-3-6-0"></a></h3>
<ul><li>Added new <code>side-widget-*</code> CSS classes to better style file download/format buttons.</li>
<li>Added <code>widget_sep_vert()</code> function.</li>
<li>Use new <code>sep-widget_*</code> CSS classes in fitting UI modules.</li>
<li>Updated report help modals text.</li>
<li>Using new <code>side-widget-*</code> CSS classes in dose estimation UI modules.</li>
<li>Added progress notifications to calculations via <code><a href="https://rdrr.io/pkg/shiny/man/Progress.html" class="external-link">shiny::Progress</a></code>.</li>
<li>Updated <code>_pkgdown.yml</code> and simplified <code>extra.css</code> thanks to Bootstrap 5 support in {pkgdown} 2.0.0.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="3.5.0" id="biodosetools-350-2021-05-26">biodosetools 3.5.0 (2021-05-26)<a class="anchor" aria-label="anchor" href="#biodosetools-350-2021-05-26"></a></h2>
<ul><li>Added unit tests for to check code coverage and to validate that the code performs as expected.</li>
<li>Major revision of the fitting and dose estimation pipelines, introducing new auxiliary functions to reduce duplicated code.</li>
<li>Major simplification of report rendering process; abandoned HTML output.</li>
<li>Added custom {pkgdown} theme for more consistent branding.</li>
<li>Added preliminary citation to README.</li>
<li>Moved packages required only for reports to <code>Suggests</code> field in <code>DESCRIPTION</code> file.</li>
</ul><div class="section level3">
<h3 id="new-features-3-5-0">New features<a class="anchor" aria-label="anchor" href="#new-features-3-5-0"></a></h3>
<ul><li>Added irradiation conditions input to dose-effect fitting modules.</li>
<li>Reports include conditional formatting of <code>u</code>-value and other formatting refinements.</li>
<li>Added <code>biodosetools_version</code> element to Shiny App’s exported <code>*-fitting-data-YYY-MM-DD.rds</code> files.</li>
<li>
<code><a href="../reference/fit.html">fit()</a></code> function allows to optionally select <code>"glm"</code> or <code>"maxlik"</code> algorithms. If <code>"glm"</code> if selected, the original <code><a href="https://rdrr.io/r/base/conditions.html" class="external-link">tryCatch()</a></code> routine will be executed.</li>
</ul></div>
<div class="section level3">
<h3 id="bug-fixes-3-5-0">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-3-5-0"></a></h3>
<ul><li>Fixed unexported <code><a href="../reference/calculate_aberr.html">calculate_aberr_var()</a></code> function.</li>
<li>Stopped using weights in fitting algorithms (<code>glm</code> and <code>glm.nb</code>). Fixes <a href="https://github.com/biodosetools-team/biodosetools/issues/20" class="external-link">#20</a>, addresses part of <a href="https://github.com/biodosetools-team/biodosetools/issues/14" class="external-link">#14</a> as well.</li>
<li>Fixed aggregated count data column parsing in <code>fix_count_data_names()</code> function.</li>
<li>Fixed case data parsing issue when <code>.csv</code> file is not perfectly formatted.</li>
<li>Fixed issue with manual fitting curve input (fixes <a href="https://github.com/biodosetools-team/biodosetools/issues/23" class="external-link">#23</a>).</li>
<li>Fixed bug in <code><a href="../reference/calculate_yield_infimum.html">calculate_yield_infimum()</a></code> where infima were being calculated for yield estimate only (fixes <a href="https://github.com/biodosetools-team/biodosetools/issues/26" class="external-link">#26</a>).</li>
<li>Fixed wrong calculation of aberrations <code>X</code> for <code>Ck</code> when <code>k&gt;=10</code> (fixes <a href="https://github.com/biodosetools-team/biodosetools/issues/27" class="external-link">#27</a>).</li>
<li>Fixed <code>fix_count_data_names()</code> to properly correct <code>Ck&gt;=C10</code> (related to issue <a href="https://github.com/biodosetools-team/biodosetools/issues/27" class="external-link">#27</a>).</li>
<li>Added “where” to global variables (see <a href="https://github.com/r-lib/tidyselect/issues/201" class="external-link uri">https://github.com/r-lib/tidyselect/issues/201</a>).</li>
</ul></div>
<div class="section level3">
<h3 id="new-functions-3-5-0">New functions<a class="anchor" aria-label="anchor" href="#new-functions-3-5-0"></a></h3>
<ul><li>
<code>inner_column()</code>, to fix padding of boxes inside columns.</li>
<li>
<code>widget_sep()</code>, to insert <code>div(class = "widget-sep", br())</code> calls.</li>
<li>
<code>widget_sep_vert()</code> to insert <code>div(style = "height: 8px;")</code> calls.</li>
<li>
<code>names_from_model_formula()</code>, to parse <code>rhandsontable()</code> Unicode row and column coefficient names.</li>
<li>
<code>parse_model_formula()</code> to get raw and TeX formulae from <code>model_formula</code>.</li>
<li>
<code><a href="../reference/calculate_trans_rate_sigurdson.html">calculate_trans_rate_sigurdson()</a></code> and <code><a href="../reference/calculate_trans_rate_manual.html">calculate_trans_rate_manual()</a></code> to calculate translocation rates.</li>
<li>
<code><a href="../reference/calculate_aberr_table.html">calculate_aberr_table()</a></code> wrapper, to calculate aberration tables for count and cases data.</li>
<li>
<code><a href="../reference/calculate_aberr.html">init_aberr_table()</a></code>, to initialise aberration distribution tables in fitting and estimation server modules.</li>
<li>
<code>list_fitting_formulas()</code>, to replace global <code>global_fitting_formulas</code> object.</li>
<li>
<code>generalise_fit_coeffs()</code> and <code>generalise_fit_var_cov_mat()</code>, used internally in estimation functions instead of using <code>general_fit_*</code> as parameters.</li>
</ul></div>
<div class="section level3">
<h3 id="function-refactoring-3-5-0">Function refactoring<a class="anchor" aria-label="anchor" href="#function-refactoring-3-5-0"></a></h3>
<ul><li>Refactored <code>get_decision_threshold()</code> function to <code>calculate_decision_threshold()</code>, and added <code>calculate_decision_threshold_table()</code> wrapper function.</li>
<li>Refactored <code>calculate_decision_threshold_*()</code> functions to remove <code>input</code> argument.</li>
<li>Refactored <code>get_*()</code> translocation functions to <code>calculate_*()</code> for better consistency.</li>
<li>Refactored <code>get_*_dose_curve()</code> functions to <code>plot_*_dose_curve()</code> for more clarity.</li>
<li>Refactored <code>get_model_statistics()</code> to <code><a href="../reference/calculate_model_stats.html">calculate_model_stats()</a></code>.</li>
<li>Refactored <code>get_fit_*()</code> functions to <code><a href="../reference/fit.html">fit()</a></code> and <code>fit_*_method()</code>.</li>
</ul></div>
<div class="section level3">
<h3 id="improvements-3-5-0">Improvements<a class="anchor" aria-label="anchor" href="#improvements-3-5-0"></a></h3>
<ul><li>Added Oliveira citation on <code>get_fit_maxlik_method()</code> function.</li>
<li>Fixed Gaëtan’s name order in citation and contributors list.</li>
<li>Updated README and About body text.</li>
<li>Renamed source R files for better naming consistency.</li>
<li>Changed structure of modules to <code>mod_&lt;calc_type&gt;_&lt;aberration&gt;_*()</code>.</li>
<li>Generalised <code>get_model_statistics()</code> function so that local <code>get_model_statistics()</code> definition could be removed from <code>mod_estimate_fit_curve_server()</code> module.</li>
<li>Multiple (18 files) <code>&lt;aberration&gt;-&lt;module&gt;-&lt;format&gt;.Rmd</code> report templates have been merged into <code>&lt;module&gt;-&lt;format&gt;.Rmd</code> (4 files).</li>
<li>Use <code><a href="../reference/correct_negative_vals.html">correct_negative_vals()</a></code> to ensure correct dose estimation when <code>X &lt; Xc</code> in translocations assay.</li>
<li>Replaced all <code>*_at()</code> and <code>*_if()</code> occurrences by their {dplyr} 1.0.0 equivalents.</li>
</ul></div>
<div class="section level3">
<h3 id="ui-improvements-3-5-0">UI Improvements<a class="anchor" aria-label="anchor" href="#ui-improvements-3-5-0"></a></h3>
<ul><li>Replaced <code>column(width = X)</code> calls by <code>col_X()</code> in UI modules.</li>
<li>Added <code>tabitem-container</code> class to <code>tabItem()</code> page containers for fixed <code>max-width</code> while keeping responsive UI.</li>
<li>Added <code>col-inner-textinput-*</code> CSS classes for Irradiation conditions <code>textInput()</code> widgets’ containers.</li>
<li>Added <code>sep-widget-download</code> and <code>sep-widget-format</code> CSS classes to unify download and format select buttons into a single widget.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="3.4.0" id="biodosetools-340-2020-10-11">biodosetools 3.4.0 (2020-10-11)<a class="anchor" aria-label="anchor" href="#biodosetools-340-2020-10-11"></a></h2>
<ul><li>Removed all non-ASCII characters.</li>
<li>Major overhaul of reports.</li>
<li>Added “Report bug” button on home screen and cleaned up <code>dashboard_*()</code> code.</li>
<li>Removed dependencies on {stringr} package.</li>
<li>Initial <code><a href="https://rdrr.io/r/utils/citation.html" class="external-link">utils::citation()</a></code> support via CITATION file.</li>
<li>Minimal version of R bumped to 3.5.0.</li>
<li>Minimal version of {dplyr} bumped to 1.0.0.</li>
</ul><div class="section level3">
<h3 id="bug-fixes-3-4-0">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-3-4-0"></a></h3>
<ul><li>Replaced <code>C</code>, <code>α</code>, <code>β</code> variables (and derivatives) by <code>coeff_C</code>, <code>coeff_alpha</code>, <code>coeff_beta</code>.</li>
<li>Fixed error in genome_fraction parsing for translocations in <code>estimate_partial_dolphin()</code>.</li>
<li>Fixed count/case data standard error column names replacements.</li>
<li>Fixed minipage issue when using more than 12 columns in {knitr} tables with column name replacements, by using <code>format = "latex", escape = FALSE</code> parameters in <code>kable()</code> call.</li>
<li>Suppress YAML warning when using <code>!expr</code> in PDF reports. See <a href="https://github.com/rstudio/rstudio/issues/7545" class="external-link uri">https://github.com/rstudio/rstudio/issues/7545</a>.</li>
</ul></div>
<div class="section level3">
<h3 id="new-functions-3-4-0">New functions<a class="anchor" aria-label="anchor" href="#new-functions-3-4-0"></a></h3>
<ul><li>
<code>fix_coeff_names()</code>, to fix coefficient names in reports.</li>
<li>
<code>fix_count_data_names()</code>, to fix count/case data column names in PDF reports.</li>
<li>
<code>to_title()</code>, to replace <code><a href="https://stringr.tidyverse.org/reference/case.html" class="external-link">stringr::str_to_title()</a></code> using base R.</li>
</ul></div>
<div class="section level3">
<h3 id="reports-improvements-3-4-0">Reports improvements<a class="anchor" aria-label="anchor" href="#reports-improvements-3-4-0"></a></h3>
<ul><li>New PDF reports, which are now the default output format.</li>
<li>Improved HTML reports styling.</li>
<li>Added explicit M-FISH usage in translocation reports.</li>
<li>Added assay name in report titles.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="3.3.1" id="biodosetools-331-2020-10-07">biodosetools 3.3.1 (2020-10-07)<a class="anchor" aria-label="anchor" href="#biodosetools-331-2020-10-07"></a></h2>
<div class="section level3">
<h3 id="bug-fixes-3-3-1">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-3-3-1"></a></h3>
<ul><li>Count and case data calculations are performed using <code>calculate_aberr_*()</code> functions. Fixes <a href="https://github.com/biodosetools-team/biodosetools/issues/8" class="external-link">#8</a>.</li>
<li>Fixed missing {dplyr} namespace in <code>n()</code> function call.</li>
<li>Fixed mismatched use of <code>awesomeCheckbox()</code> and <code>switchInput()</code> in confounders input.</li>
</ul></div>
<div class="section level3">
<h3 id="new-functions-3-3-1">New functions<a class="anchor" aria-label="anchor" href="#new-functions-3-3-1"></a></h3>
<ul><li>
<code><a href="../reference/calculate_aberr.html">calculate_aberr_power()</a></code>, which supersedes internal <code>aberr_calc()</code> function in server modules.</li>
<li>
<code><a href="../reference/calculate_aberr.html">calculate_aberr_mean()</a></code>.</li>
<li>
<code><a href="../reference/calculate_aberr.html">calculate_aberr_var()</a></code>.</li>
<li>
<code><a href="../reference/calculate_aberr.html">calculate_aberr_disp_index()</a></code>.</li>
<li>
<code><a href="../reference/calculate_aberr.html">calculate_aberr_u_value()</a></code>.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="3.3.0" id="biodosetools-330-2020-07-27">biodosetools 3.3.0 (2020-07-27)<a class="anchor" aria-label="anchor" href="#biodosetools-330-2020-07-27"></a></h2>
<p>Initial migration of the app into a {golem} R package. This means Biodose Tools is also available as a regular R package in case the user wants to build their own R scripts. This also reduced lots of code redundancies, and made the code a lot more robust.</p>
<div class="section level3">
<h3 id="new-features-3-3-0">New features<a class="anchor" aria-label="anchor" href="#new-features-3-3-0"></a></h3>
<ul><li>App is now a {golem} R package.</li>
<li>Migrated UI from {shinydashboard} to {bs4Dash}.</li>
<li>Modals are now built with {bsplus} instead of {shinyBS}.</li>
<li>Functions and internal data are documented using {roxygen2}.</li>
<li>Added {pkgdown} support to build {biodosetools}’s website automatically.</li>
<li>All custom theming is done through SASS and CSS instead of R code injection.</li>
</ul></div>
<div class="section level3">
<h3 id="new-functions-3-3-0">New functions<a class="anchor" aria-label="anchor" href="#new-functions-3-3-0"></a></h3>
<ul><li>New <code><a href="../reference/include_help.html">include_help()</a></code> for help dialogues</li>
<li>New <code>help_modal_button()</code> function to help build modal trigger buttons on boxes’ titles.</li>
<li>New <code><a href="../reference/load_rmd_report.html">load_rmd_report()</a></code> function to use in server <code>downloadHandler()</code>.</li>
<li>Moved <code>get_genome_fraction()</code> as its own exported function.</li>
</ul></div>
<div class="section level3">
<h3 id="fixes-3-3-0">Fixes<a class="anchor" aria-label="anchor" href="#fixes-3-3-0"></a></h3>
<ul><li>Fixed <a href="https://github.com/biodosetools-team/biodosetools/issues/7" class="external-link">#7</a>. Overestimate error of lambda when dispersion is smaller than 1.</li>
<li>Fixed missing icon on sidebar.</li>
<li>Fixes in collapse button paddings.</li>
<li>Fixed <code>tabBox</code> parameters.</li>
<li>Using <code>rlang::.data$</code> to avoid “no visible binding for global variable” warnings.</li>
<li>Fixed <code><a href="https://rdrr.io/pkg/bsplus/man/bs_modal.html" class="external-link">bsplus::bs_modal()</a></code> calls in <code>output$estimate_results_ui</code>.</li>
<li>Removed input as parameter in calculation functions.</li>
<li>Fixed style of help button on <code>box</code>/<code>tabBox</code> headers.</li>
<li>Updated old “biodosimetry-uab” URLs to “biodosetools-team”.</li>
</ul></div>
<div class="section level3">
<h3 id="improvements-3-3-0">Improvements<a class="anchor" aria-label="anchor" href="#improvements-3-3-0"></a></h3>
<ul><li>Stop exporting auxiliary <code>include*()</code> functions used on the app only.</li>
<li>Moved <code>widget_label()</code> to widgets.R.</li>
<li>Refactored <code>inner_column()</code> function.</li>
<li>Refactored UI modules names.</li>
<li>Refactored server functions and renamed R scripts.</li>
<li>Updated reports.</li>
<li>Using <code><a href="../reference/load_rmd_report.html">load_rmd_report()</a></code> on <code>mod_estimate_results_server()</code> module.</li>
<li>Added DNA Content Fractions of Human Chromosomes (IHGSC) data.</li>
<li>Updated <code>dna_content_fractions</code> call on <code>mod_trans_fraction_to_full_genome_server()</code> module.</li>
<li>Added support for <code>tabBox</code> on <code>help_modal_button()</code> widget.</li>
<li>Added indentation to all module <code>rhandsontable()</code> calls.</li>
<li>Stop exporting <code>get_decision_threshold()</code> function temporarily.</li>
<li>“Stains color options” boxes don’t use <code>inner_column()</code> anymore.</li>
<li>Updated <code>NEWS.md</code> headers so they can be parsed by {pkgdown}.</li>
</ul></div>
<div class="section level3">
<h3 id="ui-improvements-3-3-0">UI Improvements<a class="anchor" aria-label="anchor" href="#ui-improvements-3-3-0"></a></h3>
<ul><li>Removed <code>experiment_select</code> <code>selectInput()</code> from sidebar. Every assay is listed on the “Aberration assays” <code>sidebarMenu()</code>.</li>
<li>Loading all UI modules in <code>dashboard_body()</code>.</li>
<li>Navbar now uses <code>rightUi</code> and <code>leftUi</code> parameters, as well as <code>skin = "light"</code>.</li>
<li>Switched from <code>shinyBS::bsButton()</code> to <code><a href="https://rdrr.io/pkg/shiny/man/actionButton.html" class="external-link">shiny::actionButton()</a></code> for calling modal dialogues.</li>
<li>New CSS style sheet for {shinydashboard}.</li>
<li>Deleted old CSS files for {bs4Dash}.</li>
<li>New fonts CSS stylesheet.</li>
<li>New CSS stylesheets for fixes of existing classes and definition of custom classes.</li>
<li>Added <code>hot-improved</code> CSS class for <code>rHandsontableOutput()</code>.</li>
<li>Moved custom widgets classes from <code>theming.R</code> to <code>custom.css</code>.</li>
<li>Tweaks in <code>box</code>, <code>awesome-checkbox</code>, and sidebar submenus CSS styles.</li>
<li>Replaced <code>theme_button_status()</code> function by proper SASS to CSS compilation.</li>
<li>Finished SASS box header colors.</li>
<li>Using SASS modules to reconstruct <code>biodose_style.css</code>.</li>
<li>Moved treeview-menu style to <code>sidebar.scss</code>.</li>
<li>Added checkboxes SASS style.</li>
<li>Added missing <code>col-inner-*</code> CSS classes.</li>
<li>Added app version on sidebar footer with custom <code>sticky-footer</code> CSS class.</li>
<li>Switched to a single stylesheet, being everything built by SASS modules.</li>
<li>Added custom style to <code>radiobuttons</code>.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="3.2.1" id="biodosetools-321-2020-02-13">biodosetools 3.2.1 (2020-02-13)<a class="anchor" aria-label="anchor" href="#biodosetools-321-2020-02-13"></a></h2>
<div class="section level3">
<h3 id="bug-fixes-3-2-1">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-3-2-1"></a></h3>
<ul><li>Added required {pander} package to generate DOCX reports.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="3.2.0" id="biodosetools-320-2020-02-11">biodosetools 3.2.0 (2020-02-11)<a class="anchor" aria-label="anchor" href="#biodosetools-320-2020-02-11"></a></h2>
<p>All calculations functions previously provided in <code>inst/app/calcs</code> have been made proper functions on the package.</p>
<div class="section level3">
<h3 id="new-functions-3-2-0">New functions<a class="anchor" aria-label="anchor" href="#new-functions-3-2-0"></a></h3>
<ul><li>
<code><a href="../reference/calculate_yield.html">calculate_yield()</a></code> new wrapper of <code><a href="../reference/yield_fun.html">yield_fun()</a></code>, <code><a href="../reference/R_factor.html">R_factor()</a></code>, and <code><a href="../reference/yield_error_fun.html">yield_error_fun()</a></code>.</li>
<li>
<code><a href="../reference/calculate_yield_infimum.html">calculate_yield_infimum()</a></code> function to calculate infima of yields given a curve.</li>
<li>
<code><a href="../reference/project_yield.html">project_yield()</a></code> merged version of the <code>project_yield_estimate()</code>, <code>project_yield_lower()</code>, and <code>project_yield_upper()</code> functions.</li>
</ul></div>
<div class="section level3">
<h3 id="new-functions-3-2-0-1">“New” functions<a class="anchor" aria-label="anchor" href="#new-functions-3-2-0-1"></a></h3>
<ul><li>
<code>get_decision_threshold()</code>.</li>
<li>
<code>get_fit_dose_curve()</code>.</li>
<li>
<code>get_fit_glm_method()</code>.</li>
<li>
<code>get_fit_maxlik_method()</code>.</li>
<li>
<code>get_fit_results()</code>.</li>
<li>
<code>get_model_statistics()</code>.</li>
<li>
<code><a href="../reference/prepare_maxlik_count_data.html">prepare_maxlik_count_data()</a></code>.</li>
<li>
<code><a href="../reference/AIC_from_data.html">AIC_from_data()</a></code>.</li>
<li>
<code><a href="../reference/correct_boundary.html">correct_boundary()</a></code>.</li>
<li>
<code><a href="../reference/correct_conf_int.html">correct_conf_int()</a></code>.</li>
<li>
<code><a href="../reference/correct_negative_vals.html">correct_negative_vals()</a></code>.</li>
<li>
<code><a href="../reference/correct_yield.html">correct_yield()</a></code>.</li>
<li>
<code>get_estimated_dose_curve()</code>.</li>
<li>
<code><a href="../reference/protracted_g_function.html">protracted_g_function()</a></code>.</li>
<li>
<code><a href="../reference/R_factor.html">R_factor()</a></code>.</li>
<li>
<code><a href="../reference/yield_error_fun.html">yield_error_fun()</a></code>.</li>
<li>
<code><a href="../reference/yield_fun.html">yield_fun()</a></code>.</li>
<li>
<code>estimate_hetero()</code>.</li>
<li>
<code>estimate_partial_dolphin()</code>.</li>
<li>
<code><a href="../reference/estimate_whole_body_delta.html">estimate_whole_body_delta()</a></code>.</li>
<li>
<code>estimate_whole_body()</code>.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="3.1.0" id="biodosetools-310-2019-10-26">biodosetools 3.1.0 (2019-10-26)<a class="anchor" aria-label="anchor" href="#biodosetools-310-2019-10-26"></a></h2>
<p>Unofficial release (wasn’t changed on <code>DESCRIPTION</code> file). This includes some of the changes discussed with David in Stockholm (ERPW19).</p>
<div class="section level3">
<h3 id="new-features-3-1-0">New features<a class="anchor" aria-label="anchor" href="#new-features-3-1-0"></a></h3>
<ul><li>Made var-cov matrix optional on dose estimation inputs.</li>
<li>Fixed calculations to make var-cov matrix optional on dose estimation inputs.</li>
</ul></div>
<div class="section level3">
<h3 id="fixes-3-1-0">Fixes<a class="anchor" aria-label="anchor" href="#fixes-3-1-0"></a></h3>
<ul><li>Added additional package dependencies to <code>DESCRIPTION</code> file.</li>
<li>Following the ISO, renamed detection limits to decision threshold.</li>
</ul></div>
<div class="section level3">
<h3 id="changes-3-1-0">Changes<a class="anchor" aria-label="anchor" href="#changes-3-1-0"></a></h3>
<ul><li>Disabled decision thresholds (for now).</li>
<li>Hide AIC as a relative quality statistic of the dose estimation.</li>
<li>Added mean and variance to count data tables in fitting module.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="3.0.0" id="biodosetools-300-2019-10-12">biodosetools 3.0.0 (2019-10-12)<a class="anchor" aria-label="anchor" href="#biodosetools-300-2019-10-12"></a></h2>
<p>The app is now available as an R package on GitHub (not submitted to CRAN yet).</p>
<div class="section level3">
<h3 id="new-functions-3-0-0">New functions<a class="anchor" aria-label="anchor" href="#new-functions-3-0-0"></a></h3>
<ul><li>
<code>runApp()</code> for launching Biodose Tools.</li>
<li>
<code>%&gt;%</code> imported from {magrittr}.</li>
</ul></div>
<div class="section level3">
<h3 id="fixes-3-0-0">Fixes<a class="anchor" aria-label="anchor" href="#fixes-3-0-0"></a></h3>
<ul><li>Fixed variance calculation for count data.</li>
<li>Provide fallback method for NB2 when using constraint-maxlik-optimization.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="2.1.0" id="biodosetools-210-2019-07-27">biodosetools 2.1.0 (2019-07-27)<a class="anchor" aria-label="anchor" href="#biodosetools-210-2019-07-27"></a></h2>
<div class="section level3">
<h3 id="new-features-2-1-0">New features<a class="anchor" aria-label="anchor" href="#new-features-2-1-0"></a></h3>
<ul><li>Initial implementation of micronuclei, adding support in UI and server functions.</li>
<li>Implemented negative binomial fitting calculation for micronuclei count data.</li>
<li>Initial detection limit implementation for translocations.</li>
<li>Added DOCX support for dicentrics and translocations.</li>
</ul></div>
<div class="section level3">
<h3 id="new-functions-2-1-0">New functions<a class="anchor" aria-label="anchor" href="#new-functions-2-1-0"></a></h3>
<ul><li>New <code>bs4MyTabPanel()</code> widget to remove unnecessary padding on <code>tabCards</code>’ panels.</li>
</ul></div>
<div class="section level3">
<h3 id="fixes-2-1-0">Fixes<a class="anchor" aria-label="anchor" href="#fixes-2-1-0"></a></h3>
<ul><li>Fixed <code>colwidth</code>s for detection limits.</li>
<li>Fixed unnecessary recalculation of <code>num_cols</code> for chromosome tables in translocations modules.</li>
<li>New subdirectory structure for help dialogues.</li>
<li>Added buttons and example tables in help dialogues.</li>
<li>Updated documentation link.</li>
</ul></div>
<div class="section level3">
<h3 id="improvements-2-1-0">Improvements<a class="anchor" aria-label="anchor" href="#improvements-2-1-0"></a></h3>
<ul><li>Added <code>button_fit</code> dependency for detection limits calculation.</li>
<li>Updated fitting reports (new hot structures &amp; dynamic widths).</li>
<li>Moved calculation functions from fitting and estimation modules to their own files in a new <code>calcs</code> directory.</li>
<li>Translocations reports are complete now.</li>
</ul></div>
<div class="section level3">
<h3 id="changes-2-1-0">Changes<a class="anchor" aria-label="anchor" href="#changes-2-1-0"></a></h3>
<ul><li>Disable sourcing <code>translations.R</code> for now (until {shiny-i18n} is implemented).</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="2.0.0" id="biodosetools-200-2019-07-17">biodosetools 2.0.0 (2019-07-17)<a class="anchor" aria-label="anchor" href="#biodosetools-200-2019-07-17"></a></h2>
<div class="section level3">
<h3 id="new-features-2-0-0">New features<a class="anchor" aria-label="anchor" href="#new-features-2-0-0"></a></h3>
<ul><li>Added manual input of translocations name.</li>
<li>Implemented dynamic calculation of hot width for counts, cases, and chromosome data tables.</li>
<li>Implemented manual input of translocation frequency to be subtracted from the observed yield.</li>
<li>Started detection limit calculation in fitting module.</li>
<li>Implemented Delta method for whole-body estimation.</li>
<li>Added calculation of standard error of <code>Fg</code> for translocations’ dose estimation module.</li>
</ul></div>
<div class="section level3">
<h3 id="fixes-2-0-0">Fixes<a class="anchor" aria-label="anchor" href="#fixes-2-0-0"></a></h3>
<ul><li>Fixed dependency of the “Calculate fitting” button in the translocation module.</li>
<li>Fixed calculations for whole and partial body estimations for translocations module.</li>
<li>Fixed colors on chromosome hot table example.</li>
<li>Fix genomic conversion factor not being read when doing fitting.</li>
<li>Fix <code>Fp -&gt; y</code> when using dicentrics in dose estimation module.</li>
<li>Fixed detection limit calculation.</li>
<li>Fixed dose estimation in detection limits function.</li>
<li>Corrected correction of FISH to Full genome when inputting manual fitting curve in dose estimation modules.</li>
<li>Corrected reading of translocations fit curve values in dose estimation module.</li>
</ul></div>
<div class="section level3">
<h3 id="improvements-2-0-0">Improvements<a class="anchor" aria-label="anchor" href="#improvements-2-0-0"></a></h3>
<ul><li>Tweaked variable names and order in cases data for translocations dose estimation.</li>
<li>Updated <code>help_colors_dialog</code> for translocations modules.</li>
<li>New module for auxiliary <code>rhandsontable()</code> tables used in help dialogues.</li>
<li>New help dialogues for translocations modules.</li>
<li>Consistent use of <code>genome_fraction</code> in variables related to the genomic conversion factor/fraction of genome hybridised.</li>
<li>ggplot2 curve generation is now wrapped inside a function on the dose estimation modules.</li>
<li>Finished confounders help dialogue.</li>
<li>New aberration calculation method using <code><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">purrr::map_df()</a></code> instead of a nested loops, which is about 10 times faster.</li>
<li>Make clear in translocations dose estimation module that the coefficients are from the full genome.</li>
<li>Added highly protracted exposure.</li>
<li>Implemented calculation of detection limits from data.</li>
<li>Added dose calculation to detection limits section.</li>
<li>Changed <code>σ</code> to <code>\sigma</code> in help dialogues.</li>
<li>Added units to result tables.</li>
<li>Code clean-up on delta method for whole-body dose estimation.</li>
<li>Updated plot CI text generation on legend to account for new Delta method and for separate whole-body &amp; heterogeneous error method selection.</li>
<li>CONTRIBUTORS.md is now displayed as a modal.</li>
<li>Using ORCID for authors when possible.</li>
<li>Updated institution links in contributors files.</li>
</ul></div>
<div class="section level3">
<h3 id="ui-improvements-2-0-0">UI Improvements<a class="anchor" aria-label="anchor" href="#ui-improvements-2-0-0"></a></h3>
<ul><li>Improved <code>side-widget-tall</code> bottom margins.</li>
<li>Moved translocation frequency <code>selectInput()</code> next to Calculate Fitting button for better UX.</li>
<li>New <code>hot-improved</code> CSS class for better hot tables’ formatting.</li>
<li>All hot tables are using <code>hot-improved</code> class now.</li>
<li>New contributors list style and custom CSS style for tables.</li>
<li>Added experiment/aberration name on tab body title to avoid confusions.</li>
<li>Improved “About” tab paddings.</li>
<li>Improve widths of color columns in chromosome tables.</li>
<li>Narrower color columns in chromosome tables.</li>
<li>All hot tables’ widths are dynamic or exact now.</li>
<li>Improved small action buttons style on help dialogues.</li>
<li>New results color to differentiate from main color.</li>
<li>Updated UI to adapt for new Delta method for whole-body estimation.</li>
<li>Updated translocations UI to adapt for new Delta method for whole-body estimation.</li>
<li>Added CI for Dolphin dose estimation UI.</li>
<li>Added (work in progress) Micronuclei into UI.</li>
<li>Hide language selector until shiny-i18n is implemented.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="1.2.0" id="biodosetools-120-2019-06-25">biodosetools 1.2.0 (2019-06-25)<a class="anchor" aria-label="anchor" href="#biodosetools-120-2019-06-25"></a></h2>
<div class="section level3">
<h3 id="new-features-1-2-0">New features<a class="anchor" aria-label="anchor" href="#new-features-1-2-0"></a></h3>
<ul><li>Start translocations dose estimation module and report.</li>
<li>Allow using M-Fish color scheme in translocations fitting module.</li>
<li>Added method to convert from translocations measured by FISH to full genome.</li>
<li>Implemented confounders modifications for translocations in <code>generalEstimateCaseHotTable()</code> module.</li>
</ul></div>
<div class="section level3">
<h3 id="improvements-1-2-0">Improvements<a class="anchor" aria-label="anchor" href="#improvements-1-2-0"></a></h3>
<ul><li>New chromosome table structure. <code>transChromosomeTable()</code> and <code>transFractionToFullGenomeCalc()</code> have been moved to <code>transGeneralModule.R</code>, as they are reused in the dose estimation module.</li>
<li>Using new chromosome table UI and server modules for translocations dose estimation.</li>
<li>Merged “repeated” fitting modules into a <code>generalFittingModules.R</code> file.</li>
<li>Merged <code>*FittingResults()</code> modules into <code>generalFittingResults()</code> module.</li>
<li>Updated translocations fitting report to new chromosome table structure.</li>
<li>New <code>generalEstimateFittingCurveHotTables()</code> and <code>generalEstimateFittingCurve()</code> modules.</li>
<li>New <code>globalVariables.R</code> with <code>global_fitting_formulas</code> list.</li>
<li>Added translocations-specific code in <code>generalEstimateFittingCurve()</code> module to take into account used translocation frequency.</li>
<li>New inputs for used translocation frequency and genomic conversion factor in manual fitting input in dose estimation module.</li>
<li>Moved dose estimation to <code>generalEstimateResults()</code> module.</li>
<li>Modify translocations <code>(X - Xt)</code> for whole-body and partial-body dose estimation.</li>
<li>Added readOnly status to <code>Xt</code>, <code>yt</code>, <code>yt_err</code> variables on <code>rhandsontable()</code> for translocations case data.</li>
</ul></div>
<div class="section level3">
<h3 id="ui-improvements-1-2-0">UI improvements<a class="anchor" aria-label="anchor" href="#ui-improvements-1-2-0"></a></h3>
<ul><li>New <code>innerColumn</code> to use inside cards/boxes.</li>
<li>New <code>widgetLabel()</code> function for labels above widgets without default label.</li>
<li>Replaced old labels by new <code>widgetLabel()</code> function.</li>
<li>Finished confounders UI in translocations dose estimation module.</li>
<li>New <code>mySwichInput()</code> function based on <code><a href="https://dreamrs.github.io/shinyWidgets/reference/switchInput.html" class="external-link">shinyWidgets::switchInput()</a></code> with <code>sideLabel</code> and custom status colors.</li>
<li>Moved from <code>awesomeCheckbox()</code> to <code>mySwitchInput()</code> in all UI modules.</li>
<li>Moved fitting curve input in translocations’ dose estimation module to the top.</li>
</ul></div>
<div class="section level3">
<h3 id="fixes-1-2-0">Fixes<a class="anchor" aria-label="anchor" href="#fixes-1-2-0"></a></h3>
<ul><li>Fix minor error in calculation of infimum yield values.</li>
<li>Fixed “Color options” card and color_list object for chromosome table generation.</li>
<li>Fixed missing type argument in <code>get_model_statistics()</code> in dose estimation module.</li>
<li>Deleted heterogeneous estimation for translocations.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="1.1.0" id="biodosetools-110-2019-06-15">biodosetools 1.1.0 (2019-06-15)<a class="anchor" aria-label="anchor" href="#biodosetools-110-2019-06-15"></a></h2>
<div class="section level3">
<h3 id="new-features-1-1-0">New features<a class="anchor" aria-label="anchor" href="#new-features-1-1-0"></a></h3>
<ul><li>Conditional tabs in sidebar by implementing <code>bs4MyConditionalSidebarMenuItem()</code> function.</li>
<li>Implemented AIC calculation for different dose estimation assessments.</li>
<li>Added ZIP archive with sample data for testing.</li>
<li>New dynamic <code>bs4TabCard()</code> using <code>renderUI()</code> and <code>session$ns()</code> to show different tabs depending on user input.</li>
<li>Implemented manual calculation of case parameters (<code>N</code>, <code>X</code>, <code>y</code>, <code>y_est</code>, <code>DI</code>, <code>u</code>) in dose estimation module.</li>
</ul></div>
<div class="section level3">
<h3 id="improvements-1-1-0">Improvements<a class="anchor" aria-label="anchor" href="#improvements-1-1-0"></a></h3>
<ul><li>Disabled automatic calculation of <code>N</code>, <code>X</code>, <code>DI</code>, <code>u</code> to avoid broken input issues when running the deployed app.</li>
<li>Added <code>CONTRIBUTORS.md</code> file.</li>
<li>Removed {fontawesome} package requirement.</li>
<li>Added translocation frequency and genomic conversion factor to translocations fitting report.</li>
<li>Chromosome table included in reports.</li>
<li>Major rewrite of {ggplot2} plot in dose estimation module.</li>
<li>Added Dolphin CI info into plot when using partial-body assessment.</li>
<li>Changed display of CI in plot for dose estimation.</li>
<li>Genomic conversion factor is used in <code>transFittingResults()</code> module for calculations instead of modifying the value of <code>N</code> in <code>transFittingHotTable()</code>.</li>
<li>Added theoretical calculation of model-specific statistics (<code>glm</code> method) on fitting module.</li>
<li>Added renormalisation of data for proper calculation of model-specific statistics in fitting module.</li>
<li>Added theoretical model-specific statistics calculation for constraint <code>maxlik</code> optimization method in fitting module.</li>
<li>Added global <code>get_model_statistics()</code> to translocations fitting module.</li>
</ul></div>
<div class="section level3">
<h3 id="ui-improvements-1-1-0">UI improvements<a class="anchor" aria-label="anchor" href="#ui-improvements-1-1-0"></a></h3>
<ul><li>Cleaner version of <code>topButton()</code> for help button and help modals on fitting and dose estimation modules..</li>
<li>More informative model summary dialogue in fitting module.</li>
<li>Better output of genomic conversion factor.</li>
<li>“Partial” is now “Partial-body”. Better titles in dose estimation results card.</li>
<li>New <code>bs4MyTabCard()</code> function with topButton and noPadding capabilities.</li>
</ul></div>
<div class="section level3">
<h3 id="fixes-1-1-0">Fixes<a class="anchor" aria-label="anchor" href="#fixes-1-1-0"></a></h3>
<ul><li>Fixed problem with fit_formula_raw when using linear model. Disabled <code>*-no-int</code> models until Dolphin can deal with them.</li>
<li>Fixed rendering problem in Firefox (and potentially other browsers) with count data <code>rhandsontable()</code> using 100% of the card’s height.</li>
<li>Fixed dose estimation <code>rhandsontables()</code> <code>div</code>s on Firefox having <code>height = 100%</code> of the card’s own height.</li>
<li>Fixed height error in cases data table in dose estimation module in Firefox.</li>
<li>Fixed <code>glm_results object</code> issue in <code>get_model_statistics()</code> function.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="1.0.0" id="biodosetools-100-2019-05-24">biodosetools 1.0.0 (2019-05-24)<a class="anchor" aria-label="anchor" href="#biodosetools-100-2019-05-24"></a></h2>
<p>First public beta for laboratories and research centres.</p>
<div class="section level3">
<h3 id="new-features-1-0-0">New features<a class="anchor" aria-label="anchor" href="#new-features-1-0-0"></a></h3>
<ul><li>Initial support of reports.</li>
<li>Implemented selection of CI for dose estimation (83%-83% &amp; 95%-95%).</li>
<li>Added method selection for partial body exposure estimation.</li>
<li>Fixed formatting of variance covariance matrices when rendered by <code>rHandsontableOutput()</code>.</li>
<li>Linear models compatibility for partial-body (Dolphin) and heterogeneous dose estimation.</li>
<li>Experimental version of new Maximum Likelihood Estimation method for dose-effect fitting.</li>
<li>Manual input of fitting coefficients and var/cov matrix for dose estimation.</li>
</ul></div>
<div class="section level3">
<h3 id="improvements-1-0-0">Improvements<a class="anchor" aria-label="anchor" href="#improvements-1-0-0"></a></h3>
<ul><li>Added format/MIME type validation for file inputs.</li>
<li>Render fitting formula using MathJax.</li>
<li>Format curve depending on selected assessment/method (use appropriate CI) on dose estimation.</li>
<li>Check if yield projection into dose is mathematically possible. Negative values of yield or dose are changed to zero as well.</li>
<li>Added modification to Merkle’s yield error function when using protracted exposures.</li>
<li>Implemented protracted exposure for partial body dose estimation.</li>
<li>Added automatic Poisson/quasi-Poisson selection on fitting.</li>
<li>Correct p-values depending on fitting model dispersion.</li>
<li>Added automatic correction of confidence intervals to use “simplified” Merkle’s method if necessary.</li>
</ul></div>
<div class="section level3">
<h3 id="fixes-1-0-0">Fixes<a class="anchor" aria-label="anchor" href="#fixes-1-0-0"></a></h3>
<ul><li>Fixed calculation of full genome.</li>
<li>Fixed alignment of conditional error/method inputs for dose estimation.</li>
<li>Changed “base” to “estimate” on CI tables for dose estimation.</li>
<li>Fixed <code>bs4TabCard()</code> bug.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.2.0" id="biodosetools-020-2019-04-03">biodosetools 0.2.0 (2019-04-03)<a class="anchor" aria-label="anchor" href="#biodosetools-020-2019-04-03"></a></h2>
<p>Version presented in second Team Meeting at Munich.</p>
<div class="section level3">
<h3 id="new-features-0-2-0">New features<a class="anchor" aria-label="anchor" href="#new-features-0-2-0"></a></h3>
<ul><li>Migrated UI from {shinydashboard} to {bs4Dash}.</li>
<li>Added data, plot, and results export options.</li>
<li>Added modal dialogues using {shinyBS}.</li>
<li>Dose estimation modules for dicentrics.</li>
<li>Complete implementation of dicentrics analysis.</li>
<li>Experimental fitting for translocations assay.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.1.0" id="biodosetools-010-2018-11-29">biodosetools 0.1.0 (2018-11-29)<a class="anchor" aria-label="anchor" href="#biodosetools-010-2018-11-29"></a></h2>
<p>First draft and proof of concept presented to the RENEB team at Barcelona.</p>
<div class="section level3">
<h3 id="features-0-1-0">Features<a class="anchor" aria-label="anchor" href="#features-0-1-0"></a></h3>
<ul><li>UI built using {shinydashboard}.</li>
<li>Fitting modules for dicentrics.</li>
<li>Dynamic input tables powered by {rhandsontable}.</li>
</ul></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by <a href="https://aldomann.com" class="external-link">Alfredo Hernández</a>, David Endesfelder, Pere Puig.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer></div>

  

  

  </body></html>

