<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Changelog • biodosetools</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Changelog" />
<meta property="og:image" content="/logo.png" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-news">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">biodosetools</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">3.4.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/biodosetools-team/biodosetools/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
      <h1 data-toc-skip>Changelog <small></small></h1>
      <small>Source: <a href='https://github.com/biodosetools-team/biodosetools/blob/master/NEWS.md'><code>NEWS.md</code></a></small>
    </div>

    <div id="biodosetools-340-2020-10-11" class="section level1">
<h1 class="page-header" data-toc-text="3.4.0">
<a href="#biodosetools-340-2020-10-11" class="anchor"></a>biodosetools 3.4.0 (2020-10-11)</h1>
<ul>
<li>Removed all non-ASCII characters.</li>
<li>Major rehaul of reports.</li>
<li>Added “Report bug” button on home screen and cleaned up <code>dashboard_*()</code> code.</li>
<li>Removed dependencies on {stringr} package.</li>
<li>Initial <code><a href="https://rdrr.io/r/utils/citation.html">utils::citation()</a></code> support via CITATION file.</li>
<li>Minimal version of R bumped to 3.5.0.</li>
<li>Minimal version of {dplyr} bumped to 1.0.0.</li>
</ul>
<div id="bug-fixes" class="section level2">
<h2 class="hasAnchor">
<a href="#bug-fixes" class="anchor"></a>Bug fixes</h2>
<ul>
<li>Replaced <code>C</code>, <code>α</code>, <code>β</code> variables (and derivatives) by <code>coeff_C</code>, <code>coeff_alpha</code>, <code>coeff_beta</code>.</li>
<li>Fixed error in genome_fraction parsing for translocations in <code><a href="../reference/estimate_partial_dolphin.html">estimate_partial_dolphin()</a></code>.</li>
<li>Fixed count/case data standard error column names replacements.</li>
<li>Fixed minipage issue when using more than 12 columns in {knitr} tables with column name replacements, by using <code>format = "latex", escape = FALSE</code> parameters in <code>kable()</code> call.</li>
<li>Supress YAML warning when using <code>!expr</code> in PDF reports. See <a href="https://github.com/rstudio/rstudio/issues/7545" class="uri">https://github.com/rstudio/rstudio/issues/7545</a>.</li>
</ul>
</div>
<div id="new-functions" class="section level2">
<h2 class="hasAnchor">
<a href="#new-functions" class="anchor"></a>New functions</h2>
<ul>
<li>
<code>fix_coeff_names()</code>, to fix coefficient names in reports.</li>
<li>
<code>fix_count_data_names()</code>, to fix count/case data column names in PDF reports.</li>
<li>
<code>to_title()</code>, to replace <code><a href="https://stringr.tidyverse.org/reference/case.html">stringr::str_to_title()</a></code> using base R.</li>
</ul>
</div>
<div id="reports-improvements" class="section level2">
<h2 class="hasAnchor">
<a href="#reports-improvements" class="anchor"></a>Reports improvements</h2>
<ul>
<li>New PDF reports, which are now the default output format.</li>
<li>Improved HTML reports styling.</li>
<li>Added explicit M-FISH usage in translocation reports.</li>
<li>Added assay name in report titles.</li>
</ul>
</div>
</div>
    <div id="biodosetools-331-2020-10-07" class="section level1">
<h1 class="page-header" data-toc-text="3.3.1">
<a href="#biodosetools-331-2020-10-07" class="anchor"></a>biodosetools 3.3.1 (2020-10-07)</h1>
<div id="bug-fixes-1" class="section level2">
<h2 class="hasAnchor">
<a href="#bug-fixes-1" class="anchor"></a>Bug fixes</h2>
<ul>
<li>Count and case data calculations are performed using <code>calculate_aberr_*()</code> functions. Fixes <a href='https://github.com/biodosetools-team/biodosetools/issues/8'>#8</a>.</li>
<li>Fixed missing {dplyr} namespace in <code>n()</code> function call.</li>
<li>Fixed mismatched use of <code>awesomeCheckbox()</code> and <code>switchInput()</code> in confounders input.</li>
</ul>
</div>
<div id="new-functions-1" class="section level2">
<h2 class="hasAnchor">
<a href="#new-functions-1" class="anchor"></a>New functions</h2>
<ul>
<li>
<code><a href="../reference/calculate_aberr.html">calculate_aberr_power()</a></code>, which supersedes internal <code>aberr_calc()</code> function in server modules.</li>
<li>
<code><a href="../reference/calculate_aberr.html">calculate_aberr_mean()</a></code>.</li>
<li>
<code><a href="../reference/calculate_aberr.html">calculate_aberr_var()</a></code>.</li>
<li>
<code><a href="../reference/calculate_aberr.html">calculate_aberr_disp_index()</a></code>.</li>
<li>
<code><a href="../reference/calculate_aberr.html">calculate_aberr_u_value()</a></code>.</li>
</ul>
</div>
</div>
    <div id="biodosetools-330-2020-07-27" class="section level1">
<h1 class="page-header" data-toc-text="3.3.0">
<a href="#biodosetools-330-2020-07-27" class="anchor"></a>biodosetools 3.3.0 (2020-07-27)</h1>
</div>
    <div id="biodosetools-321-2020-02-13" class="section level1">
<h1 class="page-header" data-toc-text="3.2.1">
<a href="#biodosetools-321-2020-02-13" class="anchor"></a>biodosetools 3.2.1 (2020-02-13)</h1>
<div id="bug-fixes-2" class="section level2">
<h2 class="hasAnchor">
<a href="#bug-fixes-2" class="anchor"></a>Bug fixes</h2>
<ul>
<li>Added required {pander} package to generate DOCX reports.</li>
</ul>
</div>
</div>
    <div id="biodosetools-320-2020-02-11" class="section level1">
<h1 class="page-header" data-toc-text="3.2.0">
<a href="#biodosetools-320-2020-02-11" class="anchor"></a>biodosetools 3.2.0 (2020-02-11)</h1>
<p>All calculations functions previously provided in <code>inst/app/calcs</code> have been made proper functions on the package.</p>
<div id="new-functions-2" class="section level2">
<h2 class="hasAnchor">
<a href="#new-functions-2" class="anchor"></a>New functions</h2>
<ul>
<li>
<code><a href="../reference/calculate_yield.html">calculate_yield()</a></code> new wrapper of <code><a href="../reference/yield_fun.html">yield_fun()</a></code>, <code><a href="../reference/R_factor.html">R_factor()</a></code>, and <code><a href="../reference/yield_error_fun.html">yield_error_fun()</a></code>.</li>
<li>
<code><a href="../reference/calculate_yield_infimum.html">calculate_yield_infimum()</a></code> function to calculate infima of yields given a curve.</li>
<li>
<code><a href="../reference/project_yield.html">project_yield()</a></code> merged version of the <code>project_yield_estimate()</code>, <code>project_yield_lower()</code>, and <code>project_yield_upper()</code> functions.</li>
</ul>
</div>
<div id="new-functions-3" class="section level2">
<h2 class="hasAnchor">
<a href="#new-functions-3" class="anchor"></a>“New” functions</h2>
<ul>
<li>
<code>get_decision_threshold()</code>.</li>
<li>
<code><a href="../reference/get_fit_dose_curve.html">get_fit_dose_curve()</a></code>.</li>
<li>
<code><a href="../reference/get_fit_glm_method.html">get_fit_glm_method()</a></code>.</li>
<li>
<code><a href="../reference/get_fit_maxlik_method.html">get_fit_maxlik_method()</a></code>.</li>
<li>
<code><a href="../reference/get_fit_results.html">get_fit_results()</a></code>.</li>
<li>
<code><a href="../reference/get_model_statistics.html">get_model_statistics()</a></code>.</li>
<li>
<code><a href="../reference/prepare_maxlik_count_data.html">prepare_maxlik_count_data()</a></code>.</li>
<li>
<code><a href="../reference/AIC_from_data.html">AIC_from_data()</a></code>.</li>
<li>
<code><a href="../reference/correct_boundary.html">correct_boundary()</a></code>.</li>
<li>
<code><a href="../reference/correct_conf_int.html">correct_conf_int()</a></code>.</li>
<li>
<code><a href="../reference/correct_negative_vals.html">correct_negative_vals()</a></code>.</li>
<li>
<code><a href="../reference/correct_yield.html">correct_yield()</a></code>.</li>
<li>
<code><a href="../reference/get_estimated_dose_curve.html">get_estimated_dose_curve()</a></code>.</li>
<li>
<code><a href="../reference/protracted_g_function.html">protracted_g_function()</a></code>.</li>
<li>
<code><a href="../reference/R_factor.html">R_factor()</a></code>.</li>
<li>
<code><a href="../reference/yield_error_fun.html">yield_error_fun()</a></code>.</li>
<li>
<code><a href="../reference/yield_fun.html">yield_fun()</a></code>.</li>
<li>
<code><a href="../reference/estimate_hetero.html">estimate_hetero()</a></code>.</li>
<li>
<code><a href="../reference/estimate_partial_dolphin.html">estimate_partial_dolphin()</a></code>.</li>
<li>
<code><a href="../reference/estimate_whole_body_delta.html">estimate_whole_body_delta()</a></code>.</li>
<li>
<code><a href="../reference/estimate_whole_body.html">estimate_whole_body()</a></code>.</li>
</ul>
</div>
</div>
    <div id="biodosetools-310-2019-10-26" class="section level1">
<h1 class="page-header" data-toc-text="3.1.0">
<a href="#biodosetools-310-2019-10-26" class="anchor"></a>biodosetools 3.1.0 (2019-10-26)</h1>
<p>Unofficial release (wasn’t changed on <code>DESCRIPTION</code> file). This includes some of the changes discussed with David in Stockholm (ERPW19).</p>
<div id="new-features" class="section level2">
<h2 class="hasAnchor">
<a href="#new-features" class="anchor"></a>New features</h2>
<ul>
<li>Made var-cov matrix optional on dose estimation inputs.</li>
<li>Fixed calculations to make var-cov matrix optional on dose estimation inputs.</li>
</ul>
</div>
<div id="fixes" class="section level2">
<h2 class="hasAnchor">
<a href="#fixes" class="anchor"></a>Fixes</h2>
<ul>
<li>Added additional package dependencies to <code>DESCRIPTION</code> file.</li>
<li>Following the ISO, renamed detection limits to decision threshold.</li>
</ul>
</div>
<div id="changes" class="section level2">
<h2 class="hasAnchor">
<a href="#changes" class="anchor"></a>Changes</h2>
<ul>
<li>Disabled decision thresholds (for now).</li>
<li>Hide AIC as a relative quality statistic of the dose estimation.</li>
<li>Added mean and variance to count data tables in fitting module.</li>
</ul>
</div>
</div>
    <div id="biodosetools-300-2019-10-12" class="section level1">
<h1 class="page-header" data-toc-text="3.0.0">
<a href="#biodosetools-300-2019-10-12" class="anchor"></a>biodosetools 3.0.0 (2019-10-12)</h1>
<p>The app is now available as an R package on GitHub (we haven’t submitted Biodose Tools to CRAN yet).</p>
<div id="new-functions-4" class="section level2">
<h2 class="hasAnchor">
<a href="#new-functions-4" class="anchor"></a>New functions</h2>
<ul>
<li>
<code>runApp()</code> for launching Biodose Tools.</li>
<li>
<code><a href="../reference/%&gt;%.html">%&gt;%</a></code> imported from {magrittr}.</li>
</ul>
</div>
<div id="fixes-1" class="section level2">
<h2 class="hasAnchor">
<a href="#fixes-1" class="anchor"></a>Fixes</h2>
<ul>
<li>Fixed variance calculation for count data.</li>
<li>Provide fallback method for NB2 when using constraint-maxlik-optimization.</li>
</ul>
</div>
</div>
    <div id="biodosetools-20190727-beta" class="section level1">
<h1 class="page-header" data-toc-text="2019.07.27">
<a href="#biodosetools-20190727-beta" class="anchor"></a>biodosetools 2019.07.27-beta</h1>
<div id="new-features-1" class="section level2">
<h2 class="hasAnchor">
<a href="#new-features-1" class="anchor"></a>New features</h2>
<ul>
<li>Initial implementation of micronuclei, adding support in UI and server functions.</li>
<li>Implemented negative binomial fitting calculation for micronuclei count data.</li>
<li>Initial detection limit implementation for translocations.</li>
<li>Added DOCX support for dicentrics and translocations.</li>
</ul>
</div>
<div id="new-functions-5" class="section level2">
<h2 class="hasAnchor">
<a href="#new-functions-5" class="anchor"></a>New functions</h2>
<ul>
<li>New <code>bs4MyTabPanel()</code> widget to remove unnecessary padding on <code>tabCards</code>’ panels.</li>
</ul>
</div>
<div id="fixes-2" class="section level2">
<h2 class="hasAnchor">
<a href="#fixes-2" class="anchor"></a>Fixes</h2>
<ul>
<li>Fixed colwidths for detection limits.</li>
<li>Fixed unnecessary recalculation of <code>num_cols</code> for chromosome tables in translocations modules.</li>
<li>New subdirectory structure for help dialogues.</li>
<li>Added buttons and example tables in help dialogues.</li>
<li>Updated documentation link.</li>
</ul>
</div>
<div id="improvements" class="section level2">
<h2 class="hasAnchor">
<a href="#improvements" class="anchor"></a>Improvements</h2>
<ul>
<li>Added <code>button_fit</code> dependency for detection limits calculation.</li>
<li>Updated fitting reports (new hot structures &amp; dynamic widths).</li>
<li>Moved calculation functions from fitting and estimation modules to their own files in a new <code>calcs</code> directory.</li>
<li>Translocations reports are complete now.</li>
</ul>
</div>
<div id="changes-1" class="section level2">
<h2 class="hasAnchor">
<a href="#changes-1" class="anchor"></a>Changes</h2>
<ul>
<li>Disable sourcing translations.R for now (until {shiny-i18n} is implemented).</li>
</ul>
</div>
</div>
    <div id="biodosetools-20190717-beta" class="section level1">
<h1 class="page-header" data-toc-text="2019.07.17">
<a href="#biodosetools-20190717-beta" class="anchor"></a>biodosetools 2019.07.17-beta</h1>
</div>
    <div id="biodosetools-20190625-beta" class="section level1">
<h1 class="page-header" data-toc-text="2019.06.25">
<a href="#biodosetools-20190625-beta" class="anchor"></a>biodosetools 2019.06.25-beta</h1>
<div id="new-features-2" class="section level2">
<h2 class="hasAnchor">
<a href="#new-features-2" class="anchor"></a>New features</h2>
<ul>
<li>Start translocations dose estimation module and report.</li>
<li>Allow using M-Fish color scheme in translocations fitting module.</li>
<li>Added method to convert from translocations measured by FISH to full genome.</li>
<li>Implemented confounders modifications for translocations in <code>generalEstimateCaseHotTable()</code> module.</li>
</ul>
</div>
<div id="improvements-1" class="section level2">
<h2 class="hasAnchor">
<a href="#improvements-1" class="anchor"></a>Improvements</h2>
<ul>
<li>New chromosome table structure. <code>transChromosomeTable()</code> and <code>transFractionToFullGenomeCalc()</code> have been moved to <code>transGeneralModule.R</code>, as they are reused in the dose estimation module.</li>
<li>Using new chromosome table UI and server modules for translocations dose estimation.</li>
<li>Merged “repeated” fitting modules into a <code>generalFittingModules.R</code> file.</li>
<li>Merged <code>*FittingResults()</code> modules into <code>generalFittingResults()</code> module.</li>
<li>Updated translocations fitting report to new chromosome table structure.</li>
<li>New <code>generalEstimateFittingCurveHotTables()</code> and generalEstimateFittingCurve() modules.</li>
<li>New <code>globalVariables.R</code> with <code>global_fitting_formulas</code> list.</li>
<li>Added translocations-specific code in <code>generalEstimateFittingCurve()</code> module to take into account used translocation frequency.</li>
<li>New inputs for used translocation frequency and genomic conversion factor in manual fitting input in dose estimation module.</li>
<li>Moved dose estimation to <code>generalEstimateResults()</code> module.</li>
<li>Modify translocations <code>(X - Xt)</code> for whole-body and partial-body dose estimation.</li>
<li>Added readOnly status to <code>Xt</code>, <code>yt</code>, <code>yt_err</code> variables on <code>rhandsontable()</code> for translocations case data.</li>
</ul>
</div>
<div id="ui-improvements" class="section level2">
<h2 class="hasAnchor">
<a href="#ui-improvements" class="anchor"></a>UI improvements</h2>
<ul>
<li>New <code>innerColumn</code> to use inside cards/boxes.</li>
<li>New <code>widgetLabel()</code> function for labels above widgets without default label.</li>
<li>Replaced old labels by new <code>widgetLabel()</code> function.</li>
<li>Finished confounders UI in translocations dose estimation module.</li>
<li>New <code>mySwichInput()</code> function based on <code><a href="https://rdrr.io/pkg/shinyWidgets/man/switchInput.html">shinyWidgets::switchInput()</a></code> with <code>sideLabel</code> and custom status colors.</li>
<li>Moved from <code>awesomeCheckbox()</code> to <code>mySwitchInput()</code> in all UI modules.</li>
<li>Moved fitting curve input in translocations’ dose estimation module to the top.</li>
</ul>
</div>
<div id="fixes-3" class="section level2">
<h2 class="hasAnchor">
<a href="#fixes-3" class="anchor"></a>Fixes</h2>
<ul>
<li>Fix minor error in calculation of infimum yield values.</li>
<li>Fixed “Color options” card and color_list object for chromosome table generation.</li>
<li>Fixed missing type argument in <code><a href="../reference/get_model_statistics.html">get_model_statistics()</a></code> in dose estimation module.</li>
<li>Deleted heterogeneous estimation for translocations.</li>
</ul>
</div>
</div>
    <div id="biodosetools-20190615-beta" class="section level1">
<h1 class="page-header" data-toc-text="2019.06.15">
<a href="#biodosetools-20190615-beta" class="anchor"></a>biodosetools 2019.06.15-beta</h1>
<div id="new-features-3" class="section level2">
<h2 class="hasAnchor">
<a href="#new-features-3" class="anchor"></a>New features</h2>
<ul>
<li>Conditional tabs in sidebar by implementing <code>bs4MyConditionalSidebarMenuItem()</code> function.</li>
<li>Implemented AIC calculation for different dose estimation assessments.</li>
<li>Added ZIP archive with sample data for testing.</li>
<li>New dynamic <code>bs4TabCard()</code> using <code>renderUI()</code> and <code>session$ns()</code> to show different tabs depending on user input.</li>
<li>Implemented manual calculation of case parameters (<code>N</code>, <code>X</code>, <code>y</code>, <code>y_est</code>, <code>DI</code>, <code>u</code>) in dose estimation module.</li>
</ul>
</div>
<div id="improvements-2" class="section level2">
<h2 class="hasAnchor">
<a href="#improvements-2" class="anchor"></a>Improvements</h2>
<ul>
<li>Disabled automatic calculation of <code>N</code>, <code>X</code>, <code>DI</code>, <code>u</code> to avoid broken input issues when running the deployed app.</li>
<li>Added <code>CONTRIBUTORS.md</code> file.</li>
<li>Removed {fontawesome} package requirement.</li>
<li>Added translocation frequency and genomic conversion factor to translocations fitting report.</li>
<li>Chromosome table included in reports.</li>
<li>Major rewrite of {ggplot2} plot in dose estimation module.</li>
<li>Added Dolphin CI info into plot when using partial-body assessment.</li>
<li>Changed display of CI in plot for dose estimation.</li>
<li>Genomic conversion factor is used in <code>transFittingResults()</code> module for calculations instead of modifying the value of <code>N</code> in <code>transFittingHotTable()</code>.</li>
<li>Added theoretical calculation of model-specific statistics (<code>glm</code> method) on fitting module.</li>
<li>Added renormalisation of data for proper calculation of model-specific statistics in fitting module.</li>
<li>Added theoretical model-specific statistics calculation for constraint <code>maxlik</code> optimization method in fitting module.</li>
<li>Added global <code><a href="../reference/get_model_statistics.html">get_model_statistics()</a></code> to translocations fitting module.</li>
</ul>
</div>
<div id="ui-improvements-1" class="section level2">
<h2 class="hasAnchor">
<a href="#ui-improvements-1" class="anchor"></a>UI improvements</h2>
<ul>
<li>Cleaner version of <code>topButton()</code> for help button and help modals on fitting and dose estimation modules..</li>
<li>More informative model summary dialogue in fitting module.</li>
<li>Better output of genomic conversion factor.</li>
<li>“Partial” is now “Partial-body”. Better titles in dose estimation results card.</li>
<li>New bs4MyTabCard() function with topButton and noPadding capabilities.</li>
</ul>
</div>
<div id="fixes-4" class="section level2">
<h2 class="hasAnchor">
<a href="#fixes-4" class="anchor"></a>Fixes</h2>
<ul>
<li>Fixed problem with fit_formula_raw when using linear model. Disabled <code>*-no-int</code> models until Dolphin can deal with them.</li>
<li>Fixed rendering problem in Firefox (and potentially other browsers) with count data <code>rhandsontable()</code> using 100% of the card’s height.</li>
<li>Fixed dose estimation <code>rhandsontables()</code> <code>div</code>s on Firefox having <code>height = 100%</code> of the card’s own height.</li>
<li>Fixed height error in cases data table in dose estimation module in Firefox.</li>
<li>Fixed <code>glm_results object</code> issue in <code><a href="../reference/get_model_statistics.html">get_model_statistics()</a></code> function.</li>
</ul>
</div>
</div>
    <div id="biodosetools-20190524-beta" class="section level1">
<h1 class="page-header" data-toc-text="2019.05.24">
<a href="#biodosetools-20190524-beta" class="anchor"></a>biodosetools 2019.05.24-beta</h1>
<p>First public beta for laboratories and research centres.</p>
<div id="new-features-4" class="section level2">
<h2 class="hasAnchor">
<a href="#new-features-4" class="anchor"></a>New features</h2>
<ul>
<li>Initial support of reports.</li>
<li>Implemented selection of CI for dose estimation (83%-83% &amp; 95%-95%).</li>
<li>Added method selection for partial body exposure estimation.</li>
<li>Fixed formatting of variance covariance matrices when rendered by <code>rHandsontableOutput()</code>.</li>
<li>Linear models compatibility for partial-body (Dolphin) and heterogeneous dose estimation.</li>
<li>Experimental version of new Maximum Likelihood Estimation method for dose-effect fitting.</li>
<li>Manual input of fitting coefficients and var/cov matrix for dose estimation.</li>
</ul>
</div>
<div id="improvements-3" class="section level2">
<h2 class="hasAnchor">
<a href="#improvements-3" class="anchor"></a>Improvements</h2>
<ul>
<li>Added format/MIME type validation for file inputs.</li>
<li>Render fitting formula using MathJax.</li>
<li>Format curve depending on selected assessment/method (use appropriate CI) on dose estimation.</li>
<li>Check if yield projection into dose is mathematically possible. Negative values of yield or dose are changed to zero as well.</li>
<li>Added modification to Merkle’s yield error function when using protracted exposures.</li>
<li>Implemented protracted exposure for partial body dose estimation.</li>
<li>Added automatic poisson/quasipoison selection on fitting.</li>
<li>Correct p-values depending on fitting model dispersion.</li>
<li>Added automatic correction of confidence intervals to use “simplfied” Merkle method if necessary.</li>
</ul>
</div>
<div id="fixes-5" class="section level2">
<h2 class="hasAnchor">
<a href="#fixes-5" class="anchor"></a>Fixes</h2>
<ul>
<li>Fixed calculation of full genome.</li>
<li>Fixed alignment of conditional error/method inputs for dose estimation.</li>
<li>Changed “base” to “estimate” on CI tables for dose estimation.</li>
<li>Fixed <code>bs4TabCard()</code> bug.</li>
</ul>
</div>
</div>
    <div id="biodosetools-20190403-alpha" class="section level1">
<h1 class="page-header" data-toc-text="2019.04.03">
<a href="#biodosetools-20190403-alpha" class="anchor"></a>biodosetools 2019.04.03-alpha</h1>
<p>Version presented in second Team Meeting at Munich.</p>
<div id="new-features-5" class="section level2">
<h2 class="hasAnchor">
<a href="#new-features-5" class="anchor"></a>New features</h2>
<ul>
<li>Migrated UI from {shinydashboard} to {bs4Dash}.</li>
<li>Added data, plot, and results export options.</li>
<li>Added modal dialogues using {shinyBS}.</li>
<li>Dose estimation modules for dicentrics.</li>
<li>Complete implementation of dicentrics analysis.</li>
<li>Experimental fitting for translocations assay.</li>
</ul>
</div>
</div>
    <div id="biodosetools-20181129-alpha" class="section level1">
<h1 class="page-header" data-toc-text="2018.11.29">
<a href="#biodosetools-20181129-alpha" class="anchor"></a>biodosetools 2018.11.29-alpha</h1>
<p>First draft and proof of concept presented to the RENEB team at Barcelona.</p>
<div id="features" class="section level2">
<h2 class="hasAnchor">
<a href="#features" class="anchor"></a>Features</h2>
<ul>
<li>UI built using {shinydashboard}.</li>
<li>Fitting modules for dicentrics.</li>
<li>Dynamic input tables powered by {rhandsontable}.</li>
</ul>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>

</div>


      <footer>
      <div class="copyright">
  <p>Developed by Alfredo Hernández.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


